### PR TITLE
Fix empty line in block with EmptyStatement

### LINKED ISSUE #1361: Unstable empty line in block with EmptyStatement
```js
if (a) {
  b;


  ;
}
```
```js
if (a) {
  b;

}
```
```js
if (a) {
  b;
}
```
https://prettier.github.io/prettier/#%7B%22content%22%3A%22if%20(a)%20%7B%5Cn%20%20b%3B%5Cn%5Cn%5Cn%20%20%3B%5Cn%7D%5Cn%5Cnif%20(a)%20%7B%5Cn%20%20b%3B%5Cn%5Cn%7D%5Cn%5Cnif%20(a)%20%7B%5Cn%20%20b%3B%5Cn%7D%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%7D%7D

### PR DESCRIPTION
**Pull request by vinhlh**
Fix #1361 

cc @lydell @vjeux

### CODE REVIEW

File: src/printer.js [hunk 1]
```diff
@@ -3410,10 +3410,19 @@ function isFirstStatement(path) {
...
+  }
+
+  return stmts.slice(0, index + 1);
+}
+
 function isLastStatement(path) {
   const parent = path.getParentNode();
   const node = path.getValue();
-  const body = parent.body;
+  const body = removeEmptyStatementsOnTail(parent.body);
```
Comment (vjeux) [inline src/printer.js]: You can make it a one liner:

```js
const body = parent.body.filter(n => n.type !== "EmptyStatement");
```

**Comment by vjeux** [comment]
Thanks! If you apply my suggestion, I'll merge it :)

File: src/printer.js (hunk 1 above)
Comment (vinhlh) [inline src/printer.js]: at the first time, I thought about this.
- `filter` is more readable and maintainable.
- my current implement is a bit faster, don't have loop whole array.

it's trade-off. Let me update

**Comment by vjeux** [comment]
Thanks, right now we're not interested in extreme performance for prettier but to get it working well. We need to start profiling it to figure out what is making it slow.

Also, if you really want to make it fast, you should avoid doing allocations (stop using slice) and instead iterate from the end and go backwards until you find a node that is not an empty statement and then return with the condition.

**Pull request merged**